contracts/base/Dispatcher.sol

```diff
@@ -412,21 +386,31 @@ abstract contract Dispatcher is Payments, V2SwapRouter, V3SwapRouter, Callbacks,
         returns (bool success, bytes memory output)
     {
         // equivalent: abi.decode(inputs, (uint256, bytes, address, address, uint256, uint256))
-        uint256 value;
+        (uint256 value, bytes calldata data) = getValueAndData(inputs);
         address recipient;
         address token;
         uint256 id;
         uint256 amount;
         assembly {
-            value := calldataload(inputs.offset)
-            // 0x20 offset is the tx data, decoded below
+            // 0x00 and 0x20 offsets are value and data, above
             recipient := calldataload(add(inputs.offset, 0x40))
             token := calldataload(add(inputs.offset, 0x60))
             id := calldataload(add(inputs.offset, 0x80))
             amount := calldataload(add(inputs.offset, 0xa0))
         }
-        bytes calldata data = inputs.toBytes(1);
         (success, output) = protocol.call{value: value}(data);
         if (success) ERC1155(token).safeTransferFrom(address(this), map(recipient), id, amount, new bytes(0));
     }
+
+    /// @notice Helper function to extract `value` and `data` parameters from input bytes string
+    /// @dev The helper assumes that `value` is the first parameter, and `data` is the second
+    /// @param inputs The bytes string beginning with value and data parameters
+    /// @return value The 256 bit integer value
+    /// @return data The data bytes string
+    function getValueAndData(bytes calldata inputs) internal pure returns (uint256 value, bytes calldata data) {
```

Comment:
should we have an InputsLib? seems there are a bunch of other similar ops we run on inputs